### ISSUE TITLE
Having multiple annotations on a trait without a companion throws at runtime

### ISSUE THREAD

**Issue by kubukoz** [typelevel/cats-tagless#125]
```scala
import cats.tagless.implicits._
import cats.tagless.finalAlg
import cats.tagless.autoFunctorK
import cats.arrow.FunctionK

//or autoFunctorK + autoInstrument, or finalAlg + autoInstrument, etc.
@autoFunctorK
@finalAlg
trait Foo[F[_]] {
  def bar(a: Int): F[Int]
}

//object Foo {}

object Main extends App {
  val foo: Foo[cats.Id] = _ => 42

  println(foo.mapK(FunctionK.id))
}
```

This will result in `java.lang.NoClassDefFoundError: Foo$`. Uncommenting the companion object fixes things.

**Comment by joroKr21** [comment]
Ha, interesting. I guess the two annotations are competing on adding the companion object 🤔

**Comment by dispalt** [comment]
yup ran into this a bunch before, real annoying could never figure out a workaround.

**Comment by joroKr21** [comment]
But the tests are full of such algebras with two annotations and no companion objects:
https://github.com/typelevel/cats-tagless/blob/master/tests/src/test/scala/cats/tagless/tests/autoFunctorKTests.scala

Could it be something related to incremental compilation?
Can you try to clean and compile again?

**Comment by dispalt** [comment]
I forgot to mention but it works when its within another object, another reason why it was so annoying to figure it out.  If you move one of those traits with two annotations to the top level of the file, it won't work, IIRC.

**Comment by danslapman** [comment]
Good old bug: https://github.com/scalameta/scalameta/issues/1046, this must be fixed in scalac (since paradise was merged in 2.13)

**Comment by joroKr21** [comment]
Would this fix the problem? scala/scala#9555
